lib/pbench/server/cache_manager.py

```diff
@@ -1024,9 +1026,31 @@ def subprocess_run(
             raise exception(ctx, str(exc)) from exc
         else:
             if process.returncode != 0:
+
+                # tar failures can include a message for each file in the
+                # archive, and can be quite large. Break stderr into lines, and
+                # gather those lines only up to our configured size limit.
+                size = 0
+                message = []
+                log = logging.getLogger("sub")
+                log.info(process.stderr)
+                lines = process.stderr.split("\n")
+                for line in lines:
+                    # Skip empty lines
+                    if not line:
+                        continue
+                    size += len(line)
+                    log.info(f"LINE {line!r} ({size} <= MAX {MAX_ERROR})")
+                    if size > MAX_ERROR:
+                        break
+                    message.append(line)
+
+                # If even the first line was too big, truncate it
+                msg = "\n".join(message) if message else lines[0][:MAX_ERROR]
```

Comment:
I didn't want to truncate any output unless the first line won't fit. The size logic would have to be a bit smarter to make this work, since I'm adding in newlines "for free" (because it didn't seem worth worrying about them).

That is, if I have two lines that add to exactly `MAX_ERROR`, I'm actually storing `MAX_ERROR + 1` characters with the newline, and this would drop the last character of the second line. It frankly didn't seem worth the effort of dealing with that.

(Although, while I was typing this response, it occurred to me that it would have worked as `MAX_ERROR + len(message) - 1` ... )